### DISCUSSION TITLE
socket.io-redis-adapter createShardedAdapter usage question

### DISCUSSION THREAD

**Discussion by ghost** [socketio/socket.io#4974] [Q&A] [answered]
Hi,
I was wondering if the `createShardedAdapter` of `socket.io-redis-adapter` is meant to be used only with Redis.Cluster clients or if it does actually works with single Redis clients too. I'm asking this since I have some issues using the sharded adapter with a single Redis client

Thank you!

**Comment by darrachequesne** [comment]
It should work with single Redis clients. Could you please share the issues you are experiencing?

**Reply by ghost** [reply]
Sure!
While using a serverless sharded Redis server on Amazon Elastic Cache, I get two kinds of anomalies:

1. Using `createShardedAdapter` with *any kind of ioredis client* I always get some anomalies when using `async fetchSockets()`: some remote sockets seem to be invisible on different replicas.
2. Using `createShardedAdapter` specifically with a ioredis `Redis.Cluster` client I get almost always a `Too Many Clusters Redirects -  MOVED` error. Already tried increasing `maxRedirects` option in ioredis client, the only way to resolve this error is by increasing [`retryDelayOnMoved`](https://redis.github.io/ioredis/interfaces/ClusterOptions.html#retryDelayOnMoved) (I've not verified seriously right now though). Either ways, the issue 1 still remains.

**Reply by ghost** [reply]
Any hints on this?

**Reply by ghost** [reply]
Hi, @darrachequesne , I think maybe https://github.com/redis/ioredis/issues/1759 this can answer my questions? Do you have any updates on this matter?

**Reply by darrachequesne** [reply, marked as answer]
Yes, unfortunately https://github.com/redis/ioredis/issues/1759 indeed prevents the usage of the `ioredis` package with the sharded adapter...

**Reply by ghost** [reply]
Thank you @darrachequesne ! Do you suggest to migrate to the redis stream adapter for this use case?

**Reply by darrachequesne** [reply]
You could indeed migrate to the redis stream adapter. Please make sure to benchmark it against your specific use case, since the guarantees of both adapters are a bit different.
